File: C# Examples/08 Data Import and Export/3D Curve/SolidWorksMacro.cs
using SolidWorks.Interop.sldworks;
using System.Runtime.InteropServices;

namespace _3D_Points
{
    public partial class SolidWorksMacro
    {
        public void Main()
        {            
            ModelDoc2 swDoc = null;            
            swDoc = ((ModelDoc2)(swApp.ActiveDoc));

            swDoc.InsertCurveFileBegin();

            ExcelDataReader edr = new ExcelDataReader();
            foreach (Point p in edr.getExcelPoints(SolidWorks.Interop.swconst.swLengthUnit_e.swINCHES))
            {
                swDoc.InsertCurveFilePoint(p.X, p.Y, p.Z);
            }

            swDoc.InsertCurveFileEnd();
            swDoc.ClearSelection2(true);
            swDoc.ViewZoomtofit2();            
        }       

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 08 Data Import and Export/3DCurve/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System


Partial Class SolidWorksMacro
    Public Sub main()
        Dim swDoc As ModelDoc2 = Nothing
        swDoc = CType(swApp.ActiveDoc, ModelDoc2)
        swDoc.InsertCurveFileBegin()

        Dim edr As New ExcelDataReader2
        Dim points As List(Of Point) = edr.getExcelPoints(swLengthUnit_e.swINCHES)
        For Each pt As Point In points
            swDoc.InsertCurveFilePoint(pt.X, pt.Y, pt.Z)
        Next
        swDoc.InsertCurveFileEnd()
        swDoc.ViewZoomtofit2()
    End Sub

    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks
End Class
